### ISSUE TITLE
RelativeLayout is not taking full screen since XamarinForms 4.8 update

### ISSUE THREAD

**Issue by jpacov** [xamarin/Xamarin.Forms#11742]
### Description
Since updating to Xamarin Forms 4.8, the RelativeLayout is not taking the full space available, even when setting VerticalOptions and **HorizontalOptions** to **FillAndExpand**.
It worked fine before updating to Xamarin Forms 4.8.

### Steps to Reproduce
See code:
`

    <ContentPage.Content>
        <RelativeLayout>

            <Grid BackgroundColor="Green">
                <Grid.RowDefinitions>
                    <RowDefinition Height="*"/>
                    <RowDefinition Height="30"/>
                    <RowDefinition Height="30"/>
                </Grid.RowDefinitions>

                <Label Grid.Row="0" Text="Label1" VerticalOptions="CenterAndExpand"/>
                <Label Grid.Row="1" Text="Label2" VerticalOptions="CenterAndExpand"/>
                <Label Grid.Row="2" Text="Label3" VerticalOptions="CenterAndExpand"/>

            </Grid>
        </RelativeLayout>
    </ContentPage.Content>
`

### Expected Behavior
RelativeLayout to take full screen

### Actual Behavior
RelativeLayout not expanding to full screen, even when VerticalOptions modifiers.

### Basic Information

- Version with issue: Xamarin.Forms version: 4.8.0.1269
- Last known good version: Xamarin.Forms version: 4.7.0.1239
- IDE: VS2019
- Platform Target Frameworks: 
Only tested on:
  - Android: 9.0

### Screenshots
![image](https://user-images.githubusercontent.com/38707880/89850604-5188ae00-db50-11ea-8ccc-53a5148e8ef7.png)

### Workaround
Tried but haven't found a workaround yet

**Comment by rmarinho** [comment]
Hi @jpacov  i just tested this code with 4.6 , 4.7 and 48 and it always renders like this. 

Can you provide a sample that works fine on 4.7 and not on 4.8 ? thanks

**Comment by jpacov** [comment]
Hi @rmarinho. Sorry for that, I made a mistake when trying to simplify so much the example code.

Here is the smallest simplification I manage to reproduce the issue.
`
<ContentPage.Content>
        <RelativeLayout>
            <Grid BackgroundColor="LightGreen">
                <Grid.RowDefinitions>
                    <RowDefinition Height="*"/>
                    <RowDefinition Height="30"/>
                    <RowDefinition Height="30"/>
                </Grid.RowDefinitions>

                
                <StackLayout Grid.Row="0" BackgroundColor="LightYellow">
                    <Grid>
                        <Grid.RowDefinitions>
                            <RowDefinition Height="Auto"/>
                            <RowDefinition Height="2*"/>
                        </Grid.RowDefinitions>

                        <StackLayout Grid.Row="0" Orientation="Horizontal">
                            <Button Text="&lt;" HorizontalOptions="Start"/>
                            <Label Text="Title" HorizontalOptions="CenterAndExpand" VerticalOptions="Center"
                                    FontAttributes="Bold" FontSize="Large"/>
                            <Button Text=">" HorizontalOptions="End"/>
                        </StackLayout>

                        <CollectionView
                            Grid.Row="1"
                            Margin="10">
                            <CollectionView.ItemsLayout>
                                <LinearItemsLayout Orientation="Vertical"/>
                            </CollectionView.ItemsLayout>
                            <CollectionView.ItemsSource>
                                <x:Array Type="{x:Type x:String}">
                                    <x:String>Baboon</x:String>
                                    <x:String>Capuchin Monkey</x:String>
                                    <x:String>Blue Monkey</x:String>
                                </x:Array>
                            </CollectionView.ItemsSource>
                        </CollectionView>

                    </Grid>
                </StackLayout>

                
                <Grid Grid.Row="1" Grid.RowSpan="2" Margin="0,0" VerticalOptions="Center">
                    <Grid.ColumnDefinitions>
                        <ColumnDefinition Width="Auto"/>
                        <ColumnDefinition Width="Auto"/>
                        <ColumnDefinition Width="*"/>
                        <ColumnDefinition Width="Auto"/>
                        <ColumnDefinition Width="Auto"/>
                    </Grid.ColumnDefinitions>
                    <Frame
                        Grid.Column="0"
                        WidthRequest="60"
                        HeightRequest="60"
                        CornerRadius="30"
                        Padding="2"
                        BackgroundColor="LightGray">
                        <ImageButton
                            Source="calendar.png"
                            WidthRequest="60"
                            HeightRequest="60"/>
                    </Frame>
                </Grid>

            </Grid>
        </RelativeLayout>
    </ContentPage.Content>
`

For this I'm really sure it is working fine in version 4.7.1260, and showing a blank space in version 4.8.0.1269 as follows:
![image](https://user-images.githubusercontent.com/38707880/89948236-5c445100-dbeb-11ea-9813-dee39b11c982.png)

You can think the blank space is not so big; but believe me that in my real XAML, it is around 10 times more.

**Comment by Reveon** [comment]
I had the same issue with RelativeLayout after update to 4.8

Workaround for me was to give the child Grid a ConstraintExpression (which was not necessary for before 4.8) and set width to factor 1 like this:

```
<RelativeLayout>
     <Grid RelativeLayout.WidthConstraint="{ConstraintExpression Type=RelativeToParent, Property=Width, Factor=1, Constant=0}">
           ....
     </Grid>
</RelativeLayout>
```

**Comment by jpacov** [comment]
Thanks a lot for the workaround idea @Reveon.
By using your idea (with HeightConstraint in my case) now it worked again.

Now we leave to the Xamarin Forms experts the last decision if this workaround is **how it should be done** (which could make sense) or if it's an issue (is there an impact in performance this extra constraint?)

**Comment by rmarinho** [comment]
Hi @jpacov glad you have a workaround, but I still can't reproduce this. I tried your sample with 

4.6.0
![4 6 0](https://user-images.githubusercontent.com/1235097/90007547-d5e34a00-dc92-11ea-97d3-e11e084efb75.png)

4.7.0
![4 7 0](https://user-images.githubusercontent.com/1235097/90007570-e398cf80-dc92-11ea-9ff2-a8260ffa4613.png)

4.8.0
![4 8 0](https://user-images.githubusercontent.com/1235097/90007595-eeebfb00-dc92-11ea-91e3-cb0144f43169.png)

**Comment by Reveon** [comment]
@rmarinho 
This issue may be related to grid and I see it mostly when the content is larger then the size of the column. 
You will see the issue when you set the 1.st column size to for example 0.1* but it disapears when you make it bigger.

I have 2 grids where the top one is inside RelativeLayout. In 4.7 they are practilly identical but in 4.8 the Grid does not take the whole available space and it grows in height. See the differences on screenshots bellow.

My sample app:
[RelativeLayoutIssue.zip](https://github.com/xamarin/Xamarin.Forms/files/5069309/RelativeLayoutIssue.zip)

4.7:
![4_7](https://user-images.githubusercontent.com/29383312/90139903-42387900-dd79-11ea-934f-dec731122b1a.png)

4.8:
![4_8](https://user-images.githubusercontent.com/29383312/90139928-4d8ba480-dd79-11ea-9a57-3d4e3ae5b1d9.png)

**Comment by rmarinho** [comment]
Thanks @Reveon i can reproduce now.

**Comment by samhouts** [comment]
closed by #11903
